src/enable.js

```diff
@@ -31,48 +33,76 @@ module.exports = function (flags) {
-        log.error('enable', 'If you have just recently created this repository try running ' + chalk.yellow('greenkeeper sync'))
-        log.error('enable', 'You need admin access to a repository to enable it.')
-        log.error('enable', 'If you think this error really shouldn’t appear let us look into it with ' + chalk.yellow('greenkeeper support'))
-        process.exit(1)
-      } else if (data.statusCode === 403) {
-        log.error('enable', 'You need a paid greenkeeper.io subscription to enable private repositories\nYou can subscribe via ' + chalk.yellow('greenkeeper upgrade'))
-        process.exit(1)
-      }
-
-      log.error('enable', res.statusMessage + (res.body.message ? ': ' + res.body.message : ''))
-      process.exit(2)
-    })
+    enable(false)
+
+    function enable (isSynced) {
+      log.http('enable', 'Sending request')
+      request({
+        method: 'POST',
+        url: flags.api + 'packages',
+        json: true,
+        headers: {
+          Authorization: 'Bearer ' + flags.token
+        },
+        body: {slug: slug}
+      }, function (err, res, data) {
+        if (err) {
+          log.error('enable', err.message)
+          process.exit(2)
+        }
+
+        if (!data) {
+          return log.error('enable', story.error_no_data)
+        }
+
+        if (data.noChange) {
+          return log.error('enable', story.error_no_change(slug)) // TODO: We might not need this
+        }
+
+        if (data.ok) {
+          return console.log(story.enabled(slug))
+        }
+
+        if (data.statusCode === 400) {
+          if (!isSynced) {
+            log.verbose('enable', 'Repository not found. Starting a sync.')
+            log.info('enable', 'Synchronizing your repositories. This might take a while.')
+            var spin = spinner()
+            sync(flags, function (err, repos) {
+              clearInterval(spin)
+              if (err) {
+                log.error('enable', 'Synchronizing the repositories was not possible.')
+                return printFailedError()
+              }
+              isSynced = true
+              if (repos.indexOf(slug) > -1) {
+                log.verbose('enable', 'Repository found after sync. Trying to enable again.')
+                return enable(true)
+              } else {
+                log.verbose('enable', 'Repository not found after sync.')
+                printFailedError()
+              }
+            })
+          } else {
+            printFailedError()
+          }
+        } else if (data.statusCode === 403) {
+          log.error('enable', 'You need a paid greenkeeper.io subscription to enable private repositories\nYou can subscribe via ' + chalk.yellow('greenkeeper upgrade'))
+          process.exit(1)
+        } else {
+          log.error('enable', res.statusMessage + (res.body.message ? ': ' + res.body.message : ''))
+          process.exit(2)
+        }
+      })
```

Comment:
This is a purely stylistic preference of mine: I'm trying to avoid else cases, i.e. additional indentation levels, whenever it's possible (which is most of the time).

So whenever the previous case already returned or exited the process we can leave the else off.
By handling shorter cases earlier we can have the bigger parts of our code on a lower indentation level, and therefore it's more readable. This is what it could look like then: https://gist.github.com/boennemann/879e53de78c0e1399900bc221ed3ab89